src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -5089,14 +5089,17 @@ void attribClass(ClassSymbol c) throws CompletionFailure {
         chk.checkNonCyclic(null, c.type);
 
         Type st = types.supertype(c.type);
-        if ((c.flags_field & Flags.COMPOUND) == 0) {
+        if ((c.flags_field & Flags.COMPOUND) == 0 &&
+            (c.flags_field & Flags.SUPER_OWNER_ATTRIBUTED) == 0) {
             // First, attribute superclass.
             if (st.hasTag(CLASS))
                 attribClass((ClassSymbol)st.tsym);
```

Comment:
This looks like a guarding against a special case of attributing something twice. Did you consider pushing this into attribClass instead so that calling it a second time is a no-op?

Reply:
@jbf, sorry, I missed you comment.

I was thinking of using `Flags.UNATTRIBUTED` and skipping this whole method. I was a bit worried about a case where a class from source has a supertype from classfile (which never has `UNATTRIBUTED`), which has a supertype from source - by using `UNATTRIBUTED`, we could get to a state where this supertype attribution wouldn't attribute the super-supertype from source. Using another flag is an attempt to avoid this issue completely - the cost is using one more flag out of a limited pool of flags, of course.